src/mtype.d

```diff
@@ -8952,14 +8934,6 @@ public:
             e = e.semantic(sc);
             return e;
         }
-        if (d.parent && d.toParent().isModule())
```

Comment:
> isDataseg() returns true for only some VarDeclaration's.

I know it, but the other cases - the `e` and `d` needs to be combined with `CommaExp` - should be handled in `DotVarExp`. If its current code is not sufficient, it would be future work.